packages/di/src/lib/injectable.ts

```diff
@@ -1,52 +1,56 @@
-import { ConstructableToken } from './provider.js';
-import { INJECTABLE_MAP, Injector } from './injector.js';
-import { environment } from './environment.js';
-
-export function injectable<T extends ConstructableToken<any>>(Base: T, _?: unknown) {
-  return class InjectableNode extends Base {
-    constructor(..._: any[]) {
-      super();
-
-      // Define a new Injector and assiciate it with this instance of the service
-      const injector = new Injector(Base.providers);
-      INJECTABLE_MAP.set(this, injector);
-
-      // If the current injectable instance is a HTMLElement preform additional startup logic
-      // this will find and attach parent injectors
-      if ('HTMLElement' in globalThis && this instanceof HTMLElement) {
-        this.addEventListener('finddiroot', (e) => {
-          const parentInjector = findInjectorRoot(e);
-
-          if (parentInjector !== null) {
-            injector.setParent(parentInjector);
-          } else {
-            injector.setParent(environment());
-          }
-        });
-      }
-    }
+import { ConstructableToken, Provider } from './provider.js';
+import { Injectables, Injector } from './injector.js';
+
+export interface InjectableOpts {
+  providers: Provider<unknown>[];
+}
+
+export function injectable(opts?: InjectableOpts) {
+  return <T extends ConstructableToken<any>>(Base: T, _?: unknown) => {
+    return class InjectableNode extends Base {
```

Comment:
Just wondering if this is the best way, it does not debug well to have all classes named the same `InjectableNode`, maybe it's worth exploring other possibilities (prototype chain, proxies)!

Also wondering, if we can leave out the Custom Element hooks for injectables which won't be elements?

Reply:
If I can't find a better way to add the custom element callbacks a proxy will work well. Answer may be the prototype chain. I also want to keep this easily interoperable with other web component libraries which is why I kind of moved away from that bootstrap logic living in the element package.